### ISSUE TITLE
[NEWS] Nowe zasoby do promowania ofert / New resources to promote offers

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#3880]
Zgodnie z zapowiedzią na [stronie dla sprzedających](https://allegro.pl/dla-sprzedajacych/zmiany-dla-sprzedajacych-na-allegro-od-17-listopada-x5qPZZabdHX) **wprowadziliśmy dzisiaj zmiany w opcjach promowania ofert**:

- **dodaliśmy Wyróżnienie elastyczne** (Flexible Emphasized), dzięki któremu sprzedawca wyróżni ofertę na tyle dni, ile chce, ponieważ opłatę za nie będziemy naliczać co jeden dzień;
- **opcje Pogrubienie i Podświetlenie dostępne są tylko w Pakiecie Promo** (Promo package). Jeśli sprzedawca promuje swoje oferty przy ich pomocy, pozostaną one aktywne i widoczne dla kupujących do końca cyklu rozliczeniowego, a następnie automatycznie wyłączone;
- sprzedawca zdecyduje, czy chce wyłączyć pakiet promowania od razu, czy dopiero po zakończeniu okresu rozliczeniowego;
- opcje promowania sprzedawca może wyłączyć także w zakończonych ofertach.

Dzięki temu sprzedawca może łatwiej zarządzać opcjami promowania ofert - szybciej zareaguje na zmiany w sprzedaży, a swoje oferty wypromuje tylko wtedy, gdy uzna to za opłacalne. Więcej informacji o zmianach w promowaniu ofert znajdziesz na naszej [stronie dla sprzedających](https://allegro.pl/dla-sprzedajacych/zmiany-dla-sprzedajacych-na-allegro-od-17-listopada-x5qPZZabdHX).

**W związku ze zmianami udostępniliśmy nowe zasoby API do zarządzania promowaniem:**

- [GET /sale/offer-promotion-packages](https://developer.allegro.pl/documentation/#operation/getAvailableOfferPromotionPackages) - pobierz dostępne pakiety promowania,
- [POST /sale/offers/{offerId}/promo-options-modification](https://developer.allegro.pl/documentation/#operation/modifyOfferPromoOptionsUsingPOST) - przypisz pakiety promowania do oferty,
- [GET /sale/offers/{offerId}/promo-options](https://developer.allegro.pl/documentation/#operation/getOfferPromoOptionsUsingGET) - sprawdź pakiety promowania przypisane do oferty,
- [PUT /sale/offers/promo-options-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/promoModificationCommandUsingPUT) - przypisz pakiety promowania do wielu ofert,
- [GET /sale/offers/promo-options-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getPromoModificationCommandResultUsingGET) - pobierz wynik zadania przypisania pakietów promowania,
- [GET /sale/offers/promo-options-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getPromoModificationCommandDetailedResultUsingGET) - pobierz szczegółowy raport dla zadania przypisania pakietów promowania. 

Poniżej znajdziesz nazwy pakietów promowania w API, które przygotowaliśmy dla nowych zasobów. Są to trzy pakiety wyróżnień oraz jedna opcja dodatkowa: 

- **Flexible Emphasized** - Wyróżnienie Elastyczne, za które opłatę naliczamy codziennie;
- **Emphasized** - Wyróżnienie, za które opłatę naliczamy co dziesięć dni;
- **Promo package** - Pakiet Promo (Wyróżnienie + Podświetlenie + Pogrubienie), dla którego opłatę naliczamy codziennie;
- **Department page promo** - opcja dodatkowa, Promowanie na stronie działu, opłatę naliczamy co dziesięć dni.

**Zmiany nie mają wpływu na model oferty.** Nadal zmienisz opcję promowania za pomocą dotychczasowego API - tak jak do tej pory, zmianę wykonamy natychmiast.

Więcej informacji o tym, jak zarządzać promowaniem, znajdziesz w [naszym poradniku](https://developer.allegro.pl/my_offers/#jak-zarz%C4%85dza%C4%87-promowaniem).

-----

**Today we have introduced changes in the promo options:** 

- **we have added Flexible Emphasized**. Owing to it, the seller is able to promote an offer for as many days as he wants, because we will charge a fee for this promo option every day;
- **Bold and Highlight options are available only in the Promo package**. If at the time of implementing this change, a seller will promote his offers via those options, they will remain active and visible for buyers till the end of a cycle and then automatically disabled;
- a seller is able to decide if he wants to disable the promotion package immediately or after the end of the cycle;
- a seller can disable promo options also in ended offers.

This will make it easier for the seller to manage the offer promo options - he will react faster to changes in sales and will promote offers only when considers it profitable. You can find more information about the changes in promo options on our [for sellers page](https://allegro.pl/dla-sprzedajacych/zmiany-dla-sprzedajacych-na-allegro-od-17-listopada-x5qPZZabdHX).

Due to the changes, today we have released new API resources for managing promo options:

- [GET /sale/offer-promotion-packages](https://developer.allegro.pl/documentation/#operation/getAvailableOfferPromotionPackages) - retrieve available promo options,
- [POST /sale/offers/{offerId}/promo-options-modification](https://developer.allegro.pl/documentation/#operation/modifyOfferPromoOptionsUsingPOST) - assign promo options to an offer,
- [GET /sale/offers/{offerId}/promo-options](https://developer.allegro.pl/documentation/#operation/getOfferPromoOptionsUsingGET) - check promo options assigned to an offer,
- [PUT /sale/offers/promo-options-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/promoModificationCommandUsingPUT) - assign promo options to multiple offers,
- [GET /sale/offers/promo-options-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getPromoModificationCommandResultUsingGET) - retrieve result of the task of assigning promo options,
- [GET /sale/offers/promo-options-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getPromoModificationCommandDetailedResultUsingGET)- retrieve detailed report for the task of assigning promo options.

Below you will find the names of promo options in API, that we have prepared for the new resources. These are three packages of distinctions and one additional option:

- **Flexible Emphasized** - for which we charge a fee every day;
- **Emphasized** - for which we charge a fee every ten days;
- **Promo package** -  (Emphasized + Highlight + Bold), for which we charge a fee every day;
- **Department page promo** - additional option, for which we charge a fee every ten days;

**The change does not affect the offer model**. You will still change the promo option using the current API - as before, we will make the change immediately.

You can find more information about managing promo options in [our tutorial](https://developer.allegro.pl/en/my_offers/#how-to-manage-promo-options).

**Comment by PhotoSoft** [comment]
Czy ja dobrze rozumiem, że teraz, aby pobrać informacje o opcjach promowania danej aukcji trzeba będzie wykonać **dodatkowe** połączenie _GET /sale/offers/{offerId}/promo-options_, bo _GET /sale/offers/{offerId}_ nie zwróci nam takiej informacji?

**Comment by PrzemyslawLukanowski** [comment]
Nie, nie musisz wykonywać tego dodatkowego połączenia. W _GET /sale/offers/{offerID}_ nie zmieni się struktura oferty - w sekcji **promotion** zwrócimy odpowiednie wartości na podstawie przypisanego pakietu.

**Comment by PhotoSoft** [comment]
A przy wystawianiu nowej oferty można w tej sekcji podać nowe opcje promowania?

**Comment by PrzemyslawLukanowski** [comment]
Jeżeli ustawisz wartość true w polu emphasized, zostanie ona zmapowana do **emphasized10d** w odpowiedzi dla GET /sale/offers/{offerId}/promo-options - nowej opcji promowania emphasized1d (Flexible Emphasized) nie da się ustawić przez sekcję promotion.

**Comment by PhotoSoft** [comment]
W takim razie jest to bez sensu, bo dochodzi dodatkowe zbędne połączenie. Do tego ze zwłoką, bo trzeba czekać, aż task dot. wystawiania nowej oferty się zakończy.

To nie może tak działać...

**Comment by PhotoSoft** [comment]
Dodacie obsługę nowych opcji do sekcji promotion w sale/offers/{offerId}/ czy nie?

Głupotą jest ich obsługa tylko poprzez osobny zasób (tak jak tagów).

100 000 połączeń, aby pobrać aukcje do bazy.
100 000 połączeń, aby pobrać tagi.
I teraz dodatkowo 100 000 połączeń, aby pobrać nowe opcje promowania. Brawo!

**Comment by PrzemyslawLukanowski** [comment]
Na tę chwilę tego nie planujemy. Jeżeli zdecydujemy się wprowadzić zmiany, opublikujemy odpowiedni komunikat.

**Comment by PrzemyslawLukanowski** [comment]
@yongxiangking Możesz pobrać informację o wyróżnieniach tylko w jednej ofercie na raz - za pomocą  [GET /sale/offers/{offerId}/promo-options](https://developer.allegro.pl/documentation/#operation/getOfferPromoOptionsUsingGET).

**Comment by yongxiangking** [comment]
@PrzemyslawLukanowski  OK, jeśli chcę uzyskać więcej informacji promocyjnych, muszę zadzwonić do GET /sale/offers/{offerId}/promo-options. Interfejs ogranicza liczbę żądań, co powoduje dużo 429 (toomanyreast). Mamy nadzieję, że liczba zapytań interfejsu może być zwiększona od 9000 na minutę do więcej, lub specjalne przetwarzanie może być wykonane dla tego interfejsu.
